business/namespaces.go

```diff
@@ -391,14 +392,24 @@ func (in *NamespaceService) GetNamespacesByCluster(cluster string) ([]models.Nam
 		}
 	}
 
-	result := namespaces
+	return namespaces, nil
+}
 
-	if kialiCache != nil && in.homeClusterUserClient != nil {
-		// Set namespace by user token
-		kialiCache.SetNamespaces(in.homeClusterUserClient.GetToken(), result)
```

Comment:
I see it - this is what was wrong. We shouldn't update the cache here - this list of namespaces is incorrect because it isn't filtered.

Reply:
The only way I found to make it happen fairly consistently was to drill into node graphs, usually for service nodes.  When we drilled in we ended up making a call for istio config, which caused that sideways update, and when the timing was bad... boom.